internal/app/recordtester/recordtester_app.go

```diff
@@ -341,6 +302,111 @@ func (rt *recordTester) Start(fileName string, testDuration, pauseDuration time.
 	// <-downloader.Done()
 	// glog.Infof(`Pulling stopped after %s`, time.Since(started))
 	// exit(55, fileName, *fileArg, err)
+
+	glog.Info("Record testing done, waiting for Asset to be available")
+	if err = rt.isCancelled(); err != nil {
+		return 0, err
+	}
+
+	assetId := ""
+	wait_interval := 30 * time.Second
+	elapsed_time := 0 * time.Second
+asset:
+	for {
+		glog.Infof("Waiting %s for asset, elapsed=%s", wait_interval, elapsed_time)
+		time.Sleep(wait_interval)
+		elapsed_time = elapsed_time + wait_interval
+		if err = rt.isCancelled(); err != nil {
+			return 0, err
+		}
+
+		assets, err := rt.lapi.ListAssets() // assets are retrieved in order of most recent first
+		if err != nil {
+			glog.Errorf("Error listing assets err=%v", err)
+			return 241, err
+			// exit(241, fileName, *fileArg, err)
+		}
+		for _, asset := range *assets { // TODO: only need to check first n assets due to ordering
+			if asset.Name == fmt.Sprintf("live-to-vod-%s", sess.ID) {
```

Comment:
I think it will be even better if we start updating the `session` object with the ID of the recording asset instead. That update should be done [from here](https://github.com/livepeer/livepeer-com/blob/e5b0ec26ca996ec6cb997558bc41b25ecce7239a/packages/api/src/webhooks/cannon.ts#L516) and should be really straightforward. In a simplified way, probably a single line change as well like this:
```ts
await this.db.stream.update(sessionId, { recordingAssetId: asset.id });
```
(we'd still need to add the field to the schema, and might also wanna do that update from the outer function instead of that line I linked)

Then here instead of listing all assets, we keep `GET`ing the session until it comes with that asset ID, then we can continue from there.